src/coreclr/src/pal/src/map/virtual.cpp

```diff
@@ -2276,6 +2276,11 @@ void *ExecutableMemoryAllocator::AllocateMemoryWithinRange(const void *beginAddr
     at which the allocator should start allocating memory from its reserved memory range.
 
 --*/
+#ifdef __sun
+#define OFFSET_RAND_MAX 0x7FFFFFFF
```

Comment:
A nit - maybe add a comment that while there is RAND_MAX definition on the sunos derived OSes, it doesn't apply to the random function output.

Reply:
I have added a comment to that effect.